tests/driver_ws281x/Makefile

```diff
@@ -7,9 +7,15 @@ N ?= 8
 
 USEMODULE += ws281x
 
-ifneq (native, $(BOARD))
-  FEATURES_REQUIRED += arch_avr8
-endif
+# Currently the ws281x only supports AVR-based platforms and native
+# (via VT100 terminals).
+# See https://doc.riot-os.org/group__drivers__ws281x.html
+FEATURES_BLACKLIST += arch_arm
+FEATURES_BLACKLIST += arch_esp32
+FEATURES_BLACKLIST += arch_esp8266
+FEATURES_BLACKLIST += arch_mips32r2
+FEATURES_BLACKLIST += arch_msp430
+FEATURES_BLACKLIST += arch_riscv
```

Comment:
This is not going to work.  E.g. the GPIO ABC backend will add support for this driver on some ARM boards, but only those with GPIO ABC support.

This cannot be reasonably be solved with the current way of dependency tracking. Maybe KConfig allows to model "depends on: foo *OR* bar *OR* baz"

Reply:
Can't you just in this case do something like removing `FEATURES_BLACKLIST += arch_arm` and add `FEATURES_REQUIRES += periph_gpio_abc` for ARM platforms in `Makefile.board.dep`? And how does the current hack help with the situation?

Reply:
The current approach couldbe extended by testing for the backends one by one and adding `FEATURES_REQUIRED := arch_avr8` if none of the other backend matched.

But better would be to teach the dependency system the logical or.

Reply:
You did not answer my first question. Isn't this already solvable without hacks and the current build system approach?

Reply:
This is how the appropriate patch could look like (untested):

```diff
diff --git a/tests/driver_ws281x/Makefile b/tests/driver_ws281x/Makefile
index 43323538e9..94aaac3a38 100644
--- a/tests/driver_ws281x/Makefile
+++ b/tests/driver_ws281x/Makefile
@@ -10,7 +10,6 @@ USEMODULE += ws281x
 # Currently the ws281x only supports AVR-based platforms and native
 # (via VT100 terminals).
 # See https://doc.riot-os.org/group__drivers__ws281x.html
-FEATURES_BLACKLIST += arch_arm
 FEATURES_BLACKLIST += arch_esp32
 FEATURES_BLACKLIST += arch_esp8266
 FEATURES_BLACKLIST += arch_mips32r2
diff --git a/tests/driver_ws281x/Makefile.board.dep b/tests/driver_ws281x/Makefile.board.dep
new file mode 100644
index 0000000000..dbc03d05fd
--- /dev/null
+++ b/tests/driver_ws281x/Makefile.board.dep
@@ -0,0 +1,3 @@
+ifneq (,$(filter arch_arm,$(FEATURES_PROVIDED)))
+  FEATURES_REQUIRED += periph_gpio_abc
+endif
```

Reply:
There might be ARM boards that don't have GPIO ABC, but SPI or I2S. What really is needed is a logical or. I think that this is actually quite feasible to implement.

It would be used by e.g :

``` Makefile
FEEATURES_REQUIRED += arch_native|arch_armv8|periph_spi|periph_i2s|periph_gpio_abc
```

Reply:
Yes we currently can't do this in out `build-system`. Maybe some kind of construct similar to the one done for `FEATURES_CONFLICT` could be done. It could declare lists of options where at least one has to be provided.

```
FEATURES_REQUIRED_OPTIONS += arch_native:arch_armv8:periph_spi:periph_i2s:periph_gpio_abc
```

But this would require more reflection, and I'm not sure o the impact.